### PR TITLE
SNOW-969717: Add tests for LOB

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-lf**
### Description
Add tests for increased LOB size

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

**Review by sfc-gh-knozderko — CHANGES_REQUESTED** [review]

File: Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                {
+                    // act
+                    command.CommandText = $"{t_insertQuery} ('{C1}', '{C2}', '{C3}')";
+                    command.ExecuteNonQuery();
+
+                    command.CommandText = t_selectQuery;
+                    var reader = command.ExecuteReader();
+
+                    // assert
+                    Assert.IsTrue(reader.Read());
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs:157]: Ok

File: Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                    Assert.AreEqual(C1, reader.GetString(0));
+                    Assert.AreEqual(C2, reader.GetString(1));
+                    Assert.AreEqual(C3, reader.GetInt64(2));
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs:267]: Ok, I tried to make it similar to the other PUT/GET tests

File: Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                conn.Open();
+                using (var command = conn.CreateCommand())
+                {
+                    // Create temp table
+                    var columnNamesWithTypes = $"{s_colName[0]} VARCHAR, {s_colName[1]} VARCHAR, {s_colName[2]} INT";
+                    command.CommandText = $"CREATE OR REPLACE TABLE {t_tableName} ({columnNamesWithTypes})";
+                    command.ExecuteNonQuery();
+
+                    // Alter session result format
+                    command.CommandText = $"ALTER SESSION SET DOTNET_QUERY_RESULT_FORMAT = {_resultFormat}";
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs]: My mistake, will move the alter command

File: Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+                    Assert.AreEqual(size, row.Length);
+                }
+            }
+        }
+
+        [Test, TestCaseSource(nameof(LobSizeTestCases))]
+        public void TestLiteralInsert(int lobSize)
+        {
+            // arrange
+            string C1 = GenerateRandomString(lobSize);
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs]: Done

File: Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                    Assert.AreEqual(C3, reader.GetInt64(2));
+                }
+            }
+        }
+
+        [Test, TestCaseSource(nameof(LobSizeTestCases))]
+        public void TestPositionalInsert(int lobSize)
+        {
+            // arrange
+            string C1 = GenerateRandomString(lobSize);
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs]: Done

**Review by sfc-gh-pmotacki — APPROVED** [review]
LGTM

File: Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+            Assert.AreEqual(MaxLobSize, row[SchemaTableColumn.ColumnSize]);
+            Assert.AreEqual(SFDataType.TEXT, (SFDataType)row[SchemaTableColumn.ProviderType]);
+
+            row = dataTable.Rows[2];
+            Assert.AreEqual(s_colName[2], row[SchemaTableColumn.ColumnName]);
+            Assert.AreEqual(2, row[SchemaTableColumn.ColumnOrdinal]);
+            Assert.AreEqual(0, row[SchemaTableColumn.ColumnSize]);
+            Assert.AreEqual(SFDataType.FIXED, (SFDataType)row[SchemaTableColumn.ProviderType]);
+        }
+        private static string GenerateRandomString(int sizeInBytes)
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs:412]: let's add a new line between the methods

Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs:417]: Ok

File: Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+
+        private void CopyIntoTable(SnowflakeDbConnection conn)
+        {
+            using (var command = conn.CreateCommand())
+            {
+                // arrange
+                command.CommandText = $"COPY INTO {t_tableName}";
+                // act
+                command.ExecuteNonQuery();
+
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-lf) [reply Snowflake.Data.Tests/IntegrationTests/MaxLobSizeIT.cs:339]: I see what you mean, I've added the missing arrange/act comment

**Pull request merged**